### ISSUE TITLE
Add Medi-Cal to list of programs supported by the eligibility tool

### ISSUE THREAD

**Issue by trevorshannon** [theunitedeffort/theunitedeffort.org#403]
We may need to choose only the most common Medi-Cal eligibility routes, as it seems like there are [very many](https://ca.db101.org/ca/programs/health_coverage/medi_cal/program2.htm)

**Comment by rajivrajiv** [comment]
Hi Trevor,

You were right, Medi-Cal is a little complex. I am trying to simplify things and this is the progress so far. 
For Income-Based eligibility, I may need to find out more details about how income is counted:  
[Modified Adjusted Gross Income under the Affordable Care Act - UPDATED WITH INFORMATION FOR COVID-19 POLICIES - UC Berkeley Labor Center](https://laborcenter.berkeley.edu/modified-adjusted-gross-income-under-the-affordable-care-act/)

Would it make more sense to just stick to the FPL rules for income limits for the first release, or should we also go into details regarding how income is counted using IRS MAGI rules?


Here are some details of the 5 categories:
[DB101 California - Medi-Cal: Medi-Cal Categories](https://ca.db101.org/ca/programs/health_coverage/medi_cal/program2a.htm)
_____________________________________________________________
Medi-Cal Eligibility Categories Summary:
	Income-Based
	Aged & Disabled Federal Poverty Level
	SSI-Linked
	Medi-Cal's Working Disabled Program
	Aged, Blind & Disabled - Medically Needy

_____________________________________________________________
Common to all 5 categories:
Income Limits:
How Income is Counted:   5 different ways in 5 categories!

In addition last 4 categories:
Must Meet SSI's definition of:	Age, blindness, or disability

_____________________________________________________________
The main requirement for the 1st category is:       	  
Income Limits:			  138% of FPL for adults 19-64 years old. 266% of FPL for children 18 and under.
How Income is Counted:	   Earned and unearned income is counted based on IRS rules for modified adjusted gross income (MAGI). Learn more about what types of income affect income-based Medi-Cal eligibility.

The main requirements for the last 4 categories are:          
Must Meet SSI's definition of:	    Age, blindness, or disability

_____________________________________________________________
>Income-Based: Just 1 condition-  Income
Income Limits:			  138% of FPL for adults 19-64 years old. 266% of FPL for children 18 and under.
How Income is Counted:	   Earned and unearned income is counted based on IRS rules for modified adjusted gross income (MAGI). Learn more about what types of income affect income-based Medi-Cal eligibility.

>For age/disability categories:     (SSI eligible)*    AND   (Income limit)   AND   (Assets limit)

*This seems to be the most important one. Without this, income and assets limits do not matter.
_____________________________________________________________


Thanks for your help. I also noticed the changes made to the results and they looks really good.


Rajiv

**Comment by trevorshannon** [comment]
Many programs have a lot of complexity around what, exactly, counts as income.  The approach I've been taking is a "best effort" one.  If there are some income deductions or exclusions that can be applied without asking a ton of complicated questions in our form, then let's put those in the calculation.  A good example is a 40% deduction on self-employment income for CalFresh.  We already know which income is from self-employment, so we can easily subtract 40% of that value.  My advice is to try to figure out (or infer) the most common types of income exclusions and apply the ones that we can easily support.

However the income gets calculated, it ultimately will be compared against a FPL value.  I just found this nice [chart](https://www.coveredca.com/pdfs/FPL-chart.pdf) that helped me understand the different FPL levels used for Medi-Cal.

We can already determine SSI eligibility and age, blindness, or disability so that should not be a problem.  We do not go into the details of whether someone is disabled "enough" according to the SS Administration, as that would be to complicated and nuanced to handle in the screening form.

**Comment by rajivrajiv** [comment]
According to the chart, FPL limits for adults, children and pregnant women/individuals in the family are 138%, 266% and 213% respectively.

As a start for the 2 main types of Medi-Cal (Income-based, Disability-based), we could use 2 OR conditions for the 'may be eligible' results category:

If Income-based: 
	We could use the FPL table for 2023 and look for the 3 types of individuals in the family and apply the highest FPL limits.	
Else If Disability-based: 
	Use only the SSI eligibility condition.

**Comment by trevorshannon** [comment]
That seems like a good start.  Do you know if households qualify for Medi-cal or is it done at the individual level?  I ask because we may only have to look at the status (disabled, pregnant, child, etc) of the person filling out the form and ignore the other household members.

**Comment by rajivrajiv** [comment]
Seems like it is done at an individual level inside the family too, according to Income-based Medi-Cal:
	https://ca.db101.org/ca/programs/health_coverage/medi_cal/program2a.htm

If an adult (19-64 years) then =< 138% of FPL. (Individual level). It says nothing about a spouse, so a spouse may have to apply as an individual.
If a child (=<18 years) then =< 266% of FPL.    (Individual level).
If pregnant, then =< 213% of FPL. Unborn baby is counted as a member of the pregnant woman's family.


If income based:
	It seems like it is done on an individual basis for adults, children and pregnant people. FPL also takes into account family size to determine maximum income.

Else If disability based:	
	Aged & Disabled Federal Poverty Level		Must Meet SSI's definition of: 	Age or disability.
	SSI-Linked									                                        Age, blindness, or disability
	Medi-Cal's Working Disabled Program						                        Disability
	Aged, Blind & Disabled - Medically Needy					                        Age, blindness, or disability


SSI's definition of disability uses Substantial Gainful Activity (SGA) table uses a yearly updated data for blind and disabled people (2 values)
I am not sure if the disability option requires SSI eligibility determination, just needs checking the income limits in the SGA table for the current year's income limit. What do you think about this?


To sum up today's findings:

Income based:   (All 3 FPL options, also take into account family size)
	If child (family income =< 266% of FPL)
	If pregnant (family income =< 213% of FPL)		// unborn baby is counted as member of family.
	If Age (18-64)   AND   (income =< 138% of FPL)

Aged and disability based:
	If Age (>65)
        If Blindness (=<income limit)
        If Disability (=<income limit)


So, it seems like there are 6 main conditions. The first 3 use the FPL chart. https://www.coveredca.com/pdfs/FPL-chart.pdf
The last 2 use the SGA table.	https://ca.db101.org/ca/programs/health_coverage/medi_cal/program2a.htm (Clicking on the link for 'definition of disability', displays the table in a popup)

There are a few other eligibility considerations for disability, particularly if a person is single and living on their own, and others, which could be implemented in the 2nd pass of Medi-Cal eligibility if needed.

**Comment by trevorshannon** [comment]
Ok, doing things at the individual level is a bit easier, so that's good.  You shouldn't have to look at the checkboxes or ages for any of the household members and only use the number of household members when deciding on the income limit to compare to (as you mentioned).

I think you are right about not needing full SSI eligibility (except perhaps for SSI-linked Medi-Cal?).  You may need to refactor the `ssiCapiBaseProgram` to break out the disabled/blind/aged and substantial gainful activity bits into a separate function so it can be re-used for Medi-Cal.

Sounds like you're making good progress on this.

**Comment by rajivrajiv** [comment]
You are right about SSI-linked Medi-Cal. SSI eligibility is required for this option. For the other 3 conditions in the Aged and disability based category, the common refactored function could be used.

Income based: (All 3 FPL options, also take into account family size)
If child  AND  (family income =< 266% of FPL)
If pregnant  AND  (family income =< 213% of FPL) 		// Unborn baby is counted as member of family. Is this the same for the other eligibility programs?
If Age (18-64) AND (income =< 138% of FPL)

Aged and disability based:
IF ssiEligible AND (Age OR Blindness OR Disability)
If Age (>65)
If Blindness (=<income limit)
If Disability (=<income limit)

Next, I will work on refactoring ssiCapiBaseProgram(), with the disabled/blind/aged and SGA stuff into a separate function to be also called by medi-calProgram() and also start on medi-calProgram().

Let me know, if I have missed anything here. Thanks.

**Comment by trevorshannon** [comment]
Regarding the unborn child question, the only other program I know of that counts unborn children is WIC.  We do have a form question for that already, and it gets populated to `input.unbornChildren`

I think that 

> IF ssiEligible AND (Age OR Blindness OR Disability)

can be simplified to just **IF ssiEligible** because the age/blindness/disability is already baked into the SSI eligibility determination.

It seems unusual that there'd be a way to be eligible simply by being over age 65:

> If Age (>65)

Is that correct?

Sounds like a good plan regarding the refactor of ssiCapiBaseProgram().

**Comment by rajivrajiv** [comment]
You are correct, I will remove the standalone age condition. Age is covered by both ssiEligible and the SGA limits.

For Income based Medi-Cal, the FPL table already exists for calfreshEligible() and will reuse it.

Income based: (All 3 FPL options, also take into account family size)
If child  AND  (family income =< 266% of FPL)
If pregnant  AND  (family income =< 213% of FPL) 		// Unborn baby is counted as member of family.
If Age (18-64) AND (income =< 138% of FPL)

Aged and disability based:
If ssiEligible
If Blindness (=<income limit)
If Disability (=<income limit)

One thing I noticed in the FPL table https://www.coveredca.com/pdfs/FPL-chart.pdf.
For 138% of FPL, the table says 20121, and maybe something additional is being factored in here:
	13590 * 1.38 = 18754.2	
	13590 * 1.4806 = 20121.35

Similarly for 213% and 266%
31056 is appx. 229% and
38783 is appx. 285%

**Comment by trevorshannon** [comment]
Good catch on the calculation!  I checked all the values and noticed that only the purple-shaded columns are not consistent with the 100% FPL listed in [that table](https://www.coveredca.com/pdfs/FPL-chart.pdf).  Then I saw this further down in the document, which explains the discrepancy (last year's FPL for Covered California vs this year's FPL for Medi-Cal:

<img width="944" alt="Screenshot 2023-06-25 at 2 39 09 PM" src="https://github.com/theunitedeffort/theunitedeffort.org/assets/4656936/69961952-28b4-432d-bcb7-dada51552654">

Using the more current FPL [values](https://www.dhcs.ca.gov/services/medi-cal/eligibility/letters/Documents/23-03.pdf) linked to in that purple Medi-Cal box, the purple columns calculate out properly.

**Comment by trevorshannon** [comment]
Unassigning @rajivrajiv as we discussed separately.  You are welcome to re-assign yourself if you find time.
